proto/google/events/firebase/v1/events.proto

```diff
@@ -21,42 +21,77 @@ import "google/events/cloudevent.proto";
 
 option csharp_namespace = "Google.Events.Protobuf.Firebase.V1";
 
-// The CloudEvent raised when a document is created in the Firebase
+// The CloudEvent raised when a ref is created in the Firebase
 // Real Time Database.
-message DocumentCreated {
+message ReferenceCreated {
   option (google.events.cloud_event_type) =
-    "google.firebase.database.document.v1.created";
+    "google.firebase.database.ref.v1.created";
 
   // The data associated with the event.
-  DocumentEventData data = 1;
+  ReferenceEventData data = 1;
 }
 
-// The CloudEvent raised when a document is created in the Firebase
+// The CloudEvent raised when a reference is created in the Firebase
 // Real Time Database.
-message DocumentUpdated {
+message ReferenceUpdated {
   option (google.events.cloud_event_type) =
-    "google.firebase.database.document.v1.updated";
+    "google.firebase.database.ref.v1.updated";
 
   // The data associated with the event.
-  DocumentEventData data = 1;
+  ReferenceEventData data = 1;
 }
 
-// The CloudEvent raised when a document is created in the Firebase
+// The CloudEvent raised when a reference is created in the Firebase
 // Real Time Database.
-message DocumentDeleted {
+message ReferenceDeleted {
   option (google.events.cloud_event_type) =
-    "google.firebase.database.document.v1.deleted";
+    "google.firebase.database.ref.v1.deleted";
 
   // The data associated with the event.
-  DocumentEventData data = 1;
+  ReferenceEventData data = 1;
 }
 
-// The CloudEvent raised when a document is written (created, updated
+// The CloudEvent raised when a reference is written (created, updated
 // or deleted) in the Firebase Real Time Database.
-message DocumentWritten {
+message ReferenceWritten {
   option (google.events.cloud_event_type) =
-    "google.firebase.database.document.v1.written";
+    "google.firebase.database.ref.v1.written";
 
   // The data associated with the event.
-  DocumentEventData data = 1;
+  ReferenceEventData data = 1;
+}
+
+// The CloudEvent raised when a Remote Config is updated
+message RemoteConfigUpdated {
+  option (google.events.cloud_event_type) =
+    "google.firebase.remoteConfig.v1.updated";
+
+  // The data associated with the event.
+  RemoteConfigEventData data = 1;
+}
+
+// The CloudEvent raised when a Firebase user is created.
+message AuthUserCreated {
+  option (google.events.cloud_event_type) =
+    "google.firebase.auth.user.v1.updated";
+
+  // The data associated with the event.
+  AuthEventData data = 1;
+}
+
+// The CloudEvent raised when a Firebase user is deleted.
+message AuthUserDeleted {
```

Comment:
It feels odd to me to have `AuthEventData` be shared between both created and deleted events. What if the data were to diverge over time? Or, if this pattern is appropriate, it seems like it should behave the same as APIs where it is named after the resource it represents, not the spot it happens to be filling (i.e. `GetFoo` returns a `Foo`, not a `FooAPIData`).

Should this be something like just `UserRecord` (as in the API) or `EventUserRecord` (if we want to disambiguate) instead? Also all of the provider prefixing won't be necessary if split into multiple packages as suggested above.

Reply:
Ack for the prefixing side, and I'll bear that in mind when moving things around.

In terms of schema sharing, it's tricky - if you give each event its own schema, you can't easily use the same function for multiple events. If you use the same schema for everything, you face the exact problem you've described around divergence. The existing events *do* tend use the same schema for multiple events around the same resource... this repo is effectively modelling the existing state of the world.

As the event types have versions, it's not the end of the world to go in one direction now and change for v2 if it turns out that was a mistake. (To be clear, that's v2 just for the event types+data; it's decoupled from an "API v2" very deliberately.) That's not to underestimate the cost of adding a new major version - I don't expect this to be an *easy* thing to do, but at least it's an escape hatch so that we don't need to predict the future perfectly now.

In terms of naming, we wanted to make it really clear that this was the "data" part of the event. While the APIs often return resources, some events have data which *is* the resource (like StorageObjectData is) and others are more event-specific (e.g. Firestore and Firebase document events, where there's a delta). The eventing team is trying to work out conventions over time to make this becomes consistent.